### DISCUSSION TITLE
Whisper tokenizer is working something wrong

### DISCUSSION THREAD

**Discussion by mrmuminov** [openai/whisper#2186] [General]
![image](https://github.com/openai/whisper/assets/64766486/46890718-3f65-411b-9acb-63434f1c592a)
![image](https://github.com/openai/whisper/assets/64766486/08df3c13-bd65-400d-b13e-bda4f68e0890)

I use whiper-small tokenizer of Uzbek language. 
It's working but, something wrong.
Because, it's returned non-Uzbek characters.
How good (correct) it is?

How to fix that? 
Can I make custom tokenizer? (share some articles).

**Comment by phineas-pta** [comment]
whisper accuracy on uzbek language is very bad (WER > 90% see paper), so u may want to fine tune

custom tokenizer meaning u have to re-train whisper from scratch

**Reply by mrmuminov** [reply]
@phineas-pta

I do fine-tuning through the medum model.
But still WER is not what I expected. It is doing worse.
I think, that this poor result is due to the "wrong working" in WhisperTokenizer.

![image (3)](https://github.com/openai/whisper/assets/64766486/aa3e4e63-1740-4e08-b898-2e8f466072d3)
For example, in the image above, there are excess characters (marked with a red line).

So I was wondering if I could rewrite WhisperTokenizer. Is it really the right thing to do? I don't know if it can be done.

**Reply by phineas-pta** [reply]
fine tuning give worse WER ? 🤔 make sure your data is properly cleaned + try train longer

u can ofc create your own tokenizer but as i said it'll require re-train

**Reply by mrmuminov** [reply]
@phineas-pta 
> u can ofc create your own tokenizer but as i said it'll require re-train

I think I understand what you mean, I will try it and let you know the result.
